activerecord/test/cases/signed_id_test.rb

```diff
@@ -223,6 +227,54 @@ def set_signed_id
     Account.signed_id_verifier = old_verifier
   end
 
+  test "Account uses legacy signed ID verifier when secret is set for the model" do
+    account_signed_id_with_message_verifiers = @account.signed_id
+    assert_equal @account, Account.find_signed(account_signed_id_with_message_verifiers)
+
+    ActiveRecord.deprecator.silence do
+      Account.signed_id_verifier_secret = "account secret"
+    end
+    account_signed_id_with_secret = @account.signed_id
+    assert_equal @account, Account.find_signed(account_signed_id_with_secret)
+
+    assert_not_equal account_signed_id_with_message_verifiers, account_signed_id_with_secret
+  ensure
+    ActiveRecord.deprecator.silence do
+      Account.signed_id_verifier_secret = nil
+      Account.instance_variable_set :@signed_id_verifier, nil
+    end
+  end
+
+  test "model-specific signed_id_verifier_secret does not affect other models" do
+    toy_signed_id = @toy.signed_id
+
+    ActiveRecord.deprecator.silence do
+      Account.signed_id_verifier_secret = "account secret"
+    end
+
+    assert_equal @toy.signed_id, toy_signed_id
+  ensure
+    ActiveRecord.deprecator.silence do
+      Account.signed_id_verifier_secret = nil
+      Account.instance_variable_set :@signed_id_verifier, nil
+    end
+  end
```

Comment:
Hmm, I don't this test is necessary.  I think it is sufficient to test that `ActiveRecord::Base.signed_id_verifier` is not affected (which we do in another test).  Is there a case I am missing?

Reply:
No you're right. Checking `ActiveRecord::Base.signed_id_verifier` can be enough.